port_ocean/utils/cache.py

```diff
@@ -0,0 +1,32 @@
+import functools
+from typing import Callable, AsyncIterator, Any
+
+from port_ocean.context.event import event
+
+AsyncIteratorCallable = Callable[..., AsyncIterator[list[Any]]]
+
+
+def cache_iterator_result(
+    cache_key: str,
+) -> Callable[[AsyncIteratorCallable], AsyncIteratorCallable]:
```

Comment:
if we want this to be user by other integration we should add it a proper logs on how it should be used